scipy/special/_orthogonal.py

```diff
@@ -1484,8 +1490,18 @@ def roots_gegenbauer(n, alpha, mu=False):
         # keep doing so.
         return roots_chebyt(n, mu)
 
-    mu0 = (np.sqrt(np.pi) * _ufuncs.gamma(alpha + 0.5)
-           / _ufuncs.gamma(alpha + 1))
+    if alpha <= 170:
+        mu0 = (np.sqrt(np.pi) * _ufuncs.gamma(alpha + 0.5)) \
+              / _ufuncs.gamma(alpha + 1)
+    else:
+        inv_rt_alpha = 1. / np.sqrt(alpha)
+        mu0 = np.sqrt(np.pi) * (inv_rt_alpha
```

Comment:
Two things here:

- Let's make sure we're adding commentary on what this asymptotic approximation is
- You want to be using Horner's method to evaluate this polynomial

Reply:
Great ideas, thanks @person142! The Horner's method version is faster and slightly more accurate. I also added a couple quick comments to the other approximations. New changes are pushed.